service/dps/utils/src/main/java/eu/europeana/cloud/service/dps/service/utils/validation/DpsTaskValidator.java

```diff
@@ -183,10 +204,19 @@ public void validate(DpsTask task) throws DpsTaskValidationException {
                 validateId(task, re);
             } else if (fieldType.equals(DpsTaskFieldType.OUTPUT_REVISION)) {
                 validateOutputRevision(task, revisionMustExist);
+            } else if (fieldType.equals(DpsTaskFieldType.TARGET_INDEXING_DATABASE)) {
+                validateTargetIndexingDatabase(task, re);
             }
         }
     }
 
+    private void validateTargetIndexingDatabase(DpsTask task, DpsTaskConstraint constraint) throws DpsTaskValidationException {
```

Comment:
I don't think we shoud have a seperate method to validate TARGET_INDEXING_DATABASE. It is just a parameter so we should add another paramter validator to validate it against multiple values (in this case PUBLISH,PREVIEW):
STH like:
  public DpsTaskValidator withParameter(String parameterName, List<String> values)
{
}
and then handle the generated constaint!